VotingApplication/VotingApplication.Web.Api/Scripts/BasicVote.js

```diff
@@ -152,47 +83,62 @@
 
         self.displayResults = function (data) {
             var groupedVotes = countVotes(data);
-            drawChart(groupedVotes);
-        }
+            self.drawChart(groupedVotes);
+        };
+
+        self.initialise = function (pollData) {
+            self.pollOptions.initialise(pollData);
+        };
 
-        self.addOption = function () {
-            //Don't submit without an entry in the name field
-            if ($("#newName").val() === "") {
+        // TODO: Extract chart code from viewModel class - ideally
```

Comment:
Any particular reason? I moved the code to the end and added the comment to make it clear that it wasn't, and shouldn't be, part of the viewModel code. Personally I think it's valid to keep that comment there until the problem is resolved.